src/components/CreateArticle.jsx

```diff
@@ -0,0 +1,214 @@
+    if ((this.props.publish && prevProps.publish === false)
+    || (this.props.draft && prevProps.draft === false)) {
+      let uploaded = true;
+      const { onSave, createArticle } = this.props;
+      const {
+        photo,
+        image,
+        id,
+        title,
+        body
+      } = prevState;
+      if (title.trim().length < 3 || body.trim().length < 3) {
+        onSave();
+        this.props.addFlashMessage({
+          type: 'error',
+          text: 'title and body must not be less than 3 characters'
+        });
+      } else {
+        if (photo && image) {
+          uploaded = await uploadImageCloudinary(image);
+        }
+        if (uploaded) {
+          uploaded = ((typeof uploaded) === 'string') ? uploaded : null;
+          // eslint-disable-next-line react/no-did-update-set-state
+          this.setState({ image: uploaded }, async () => {
+            if (id) {
+            //
+            } else {
+              onSave();
+              const article = await createArticle(this.state);
+              if (!image) {
+                this.onClose();
+              }
+              if (article.success) {
+                this.setState({
+                  id: article.data.id,
+                  title: article.data.title,
+                  description: article.data.description,
+                  body: article.data.body,
+                  readTime: article.data.readTime,
+                  image: article.data.image
+                });
+                this.props.addFlashMessage({
+                  type: 'success',
+                  text: 'Article saved to draft'
+                });
+              } else {
+                this.props.addFlashMessage({
+                  type: 'error',
+                  text: article.message
+                });
+              }
+            }
+          });
+        } else {
+          this.props.addFlashMessage({
+            type: 'error',
+            text: 'failed to upload image'
+          });
+        }
+      }
+    }
+  }
+
+  onChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+    const scroll = event.target.scrollHeight;
+    event.target.style.height = `${scroll}px`;
+  }
+
+  inputChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  fileChange = (event) => {
+    const file = event.target.files[0];
+    this.setState({
+      [event.target.name]: event.target.files[0]
+    });
+    this.readFile(file);
+  }
+
+  readFile = (file) => {
+    if (file) {
+      // eslint-disable-next-line no-undef
+      const FR = new FileReader();
+
+      FR.addEventListener('load', (event) => {
+        this.setState({
+          image: event.target.result
+        });
+      });
+
+      FR.readAsDataURL(file);
+    }
+  }
+
+  onImageClick = () => {
+    document.getElementById('add-image').click();
+  }
+
+  onClose = () => {
+    this.setState({
+      image: null
```

Comment:
Why are we doing this?

Reply:
This is required to be able to remove an image if the author at a point decides it's not necessary